lib/maths/CSignal.cc

```diff
@@ -13,6 +13,7 @@
 #include <boost/math/constants/constants.hpp>
 
 #include <algorithm>
+#include <cinttypes>
```

Comment:
It should probably be `<cstdint>` instead of `<cinttypes>` - the former contains the types and the latter adds `printf` macros, which we aren't using here.

Reply:
Good catch. My version of clangd occasionally seems to add this header for no good reason. See [37658a9](https://github.com/elastic/ml-cpp/pull/1267/commits/37658a903a834b0f03630de9675c396bf3187beb).